tests/container_checker/test_container_checker.py

```diff
@@ -0,0 +1,299 @@
+
+
+def postcheck_critical_processes_status(duthost, up_bgp_neighbors):
+    """Calls the functions to post-check the status of critical processes and
+       state of BGP sessions.
+
+    Args:
+        duthost: Host DUT.
+        skip_containers: A list contains the container names which should be skipped.
+
+    Return:
+        If all critical processes are running and all BGP sessions are established, it
+        returns True. Otherwise it will call the function to do post-check every 30 seconds
+        for 3 minutes. It will return False after timeout
+    """
+    return wait_until(CONTAINER_RESTART_THRESHOLD_SECS, CONTAINER_CHECK_INTERVAL_SECS,
+                      post_test_check, duthost, up_bgp_neighbors)
+
+
+def stop_containers(duthost, container_autorestart_states, skip_containers):
+    """Stops the running containers and returns their names .
+
+    Args:
+        duthost: Host DUT.
+        container_autorestart_states: A dictionary which key is container name and
+        value is the state of autorestart feature.
+        skip_containers: A list contains the container names which should be skipped.
+
+    Return:
+        A list which contains the container names which are stopped to run.
+    """
+    stopped_containers_list = []
+
+    for container_name in container_autorestart_states.keys():
+        if container_name not in skip_containers:
+            logger.info("Stopping the container '{}'...".format(container_name))
+            duthost.shell("sudo systemctl stop {}.service".format(container_name))
+            logger.info("Waiting until container '{}' is stopped...".format(container_name))
+            stopped = wait_until(CONTAINER_STOP_THRESHOLD_SECS,
+                                 CONTAINER_CHECK_INTERVAL_SECS,
+                                 check_container_state, duthost, container_name, False)
+            pytest_assert(stopped, "Failed to stop container '{}'".format(container_name))
+            logger.info("Container '{}' was stopped".format(container_name))
+            stopped_containers_list.append(container_name)
+
+    return stopped_containers_list
+
+
+def check_alerting_message(duthost, stopped_containers_list):
+    """Checks whether the names of stopped containers appear in Monit alerting message.
+
+    Args:
+        duthost: Host DUT.
+        stopped_containers_list: A list of stopped container names.
+
+    Return:
+        None.
+    """
+    alerting_message = duthost.shell("sudo cat /var/log/syslog | grep -m 1 '.*monit.*container_checker'",
+                                     module_ignore_errors=True)
+
+    pytest_assert(len(alerting_message["stdout_lines"]) > 0,
+                  "Failed to get Monit alerting message from container_checker!")
+
+    for container_name in stopped_containers_list:
+        if container_name not in alerting_message["stdout_lines"][0]:
```

Comment:
Why ```alerting_message["stdout_lines"][0]``` ? Always in the first line?

Reply:
At here, after the running containers except `database`  were stopped for 6 minutes, the script `container_checker` would write the alerting message into the syslog. So at the line 223, I used the command `sudo cat /var/log/syslog | grep -m 1 '.*monit.*container_checker'` to try to get the message generated by `container_checker`.  Since this command will only get a single alerting message from syslog, that's why at there I used ` alerting_message["stdout_lines"][0]`. I do not know whether such method is appropriate or not? Any suggestions?

Reply:
Updated with a flexible method! Please review again!

Reply:
I see. Thanks!